x/emissions/keeper/topic_activation.go

```diff
@@ -157,6 +157,32 @@ func (k *Keeper) inactivateTopicWithoutMinWeightReset(ctx context.Context, topic
 		return errors.Wrap(err, "failed to remove active topics")
 	}
 
+	// Remove previous weight from total sum of previous topic weights, but keep on list
```

Comment:
so this is good, but shouldn't there be an equivalent function happening when a topic is first activated? is that happening in the keeper and i'm not seeing it?

Reply:
Good catch. 
This is taken care of in the setPreviousTopicWeight for new topics that get activated, but old topics with a weight should also have this protection or we would be double-adding to total weights.
I'm reviewing this now. 
One of the candidates for explicit unit testing: activate/inactivate/activate, and check total weights.

Reply:
Fixed, tested with current tests, but pending adding the test defined above.
No test was breaking before or after this.

Reply:
added test for such case